sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/ScriptTransformationSuite.scala

```diff
@@ -228,6 +228,26 @@ class ScriptTransformationSuite extends SparkPlanTest with SQLTestUtils with Tes
         'e.cast("string")).collect())
     }
   }
+
+  test("SPARK-30973: ScriptTransformationExec should wait for the termination of script") {
+    (0 until 10).foreach { _ =>
```

Comment:
The loop is not needed.

Reply:
This corner case does not happen every time,  so that I add this loop to reproduce this problem.

Reply:
I know.  But even with 10 times loop it's  still possible that the issue can not be reproduced, right?

Reply:
OK, the loop is deleted.